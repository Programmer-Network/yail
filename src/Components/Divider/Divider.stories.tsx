import type { Meta, StoryObj } from "@storybook/react-vite";

import Divider from ".";

const meta = {
  title: "Core/Divider",
  component: Divider,
  parameters: {
    layout: "centered"
  },
  tags: ["autodocs"],
  argTypes: {}
} satisfies Meta<typeof Divider>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {}
};

export const WithClassesProperty: Story = {
  args: {
    classes: ["yl:bg-violet-300", "yl:bg-violet-400", "yl:bg-violet-500"]
  }
};
