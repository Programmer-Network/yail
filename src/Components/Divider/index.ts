export { default } from "./Divider";
export * from "./types";
