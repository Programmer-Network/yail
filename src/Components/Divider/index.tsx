import classNames from "classnames";
import { FC } from "react";

import { ArrayUtils } from "../../Utils";
import { IDividerProps } from "./types";

const Divider: FC<IDividerProps> = (
  { className, classes } = {
    classes: ["bg-text", "bg-text", "bg-text"]
  }
) => {
  if (classes && !Array.isArray(classes)) {
    throw new Error('ðŸ’£ Divider: "classes" must be an array ðŸ’£ of class names');
  }

  return (
    <div
      data-testid='divider'
      className={classNames("mx-auto flex justify-center gap-1", className)}
    >
      {Array.isArray(classes) &&
        ArrayUtils.shuffle(classes).map((bg, index) => (
          <div
            key={index}
            className={classNames("block h-[2px] w-28", bg)}
            data-testid={`divider-child-${index}`}
          />
        ))}
    </div>
  );
};

export default Divider;
