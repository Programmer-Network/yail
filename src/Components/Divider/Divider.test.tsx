import { render, screen } from "@testing-library/react";

import Divider from "./";

describe("Divider component", () => {
  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(
      <Divider classes={["yl:bg-text", "yl:bg-text", "yl:bg-text"]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("applies custom class name", () => {
    const customClass = "custom-class";
    render(
      <Divider
        className={customClass}
        classes={["yl:bg-text", "yl:bg-text", "yl:bg-text"]}
      />
    );
    const divider = screen.getByTestId("divider");
    expect(divider).toHaveClass(customClass);
  });

  test("throws error if classes prop is not an array", () => {
    const invalidClasses = "invalid-classes";
    // @ts-expect-error - Passing invalid classes prop to Divider component for testing purposes
    expect(() => render(<Divider classes={invalidClasses} />)).toThrow(
      'ğŸ’£ Divider: "classes" must be an array ğŸ’£ of class names'
    );
  });
});
