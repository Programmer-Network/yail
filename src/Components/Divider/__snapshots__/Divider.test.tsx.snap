// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mx-auto flex justify-center gap-1"
    data-testid="divider"
  >
    <div
      class="block h-[2px] w-28 bg-text"
      data-testid="divider-child-0"
    />
    <div
      class="block h-[2px] w-28 bg-text"
      data-testid="divider-child-1"
    />
    <div
      class="block h-[2px] w-28 bg-text"
      data-testid="divider-child-2"
    />
  </div>
</DocumentFragment>
`;
